crates/circuit/src/packed_instruction.rs

```diff
@@ -628,3 +729,23 @@ impl PackedInstruction {
         }
     }
 }
+
+// When cloning a `PackedInstruction` we will need to delete the old cached
+// `Gate` to avoid having duplicated references to the same python object.
+impl Clone for PackedInstruction {
+    fn clone(&self) -> Self {
+        Self {
+            op: self.op.clone(),
+            qubits: self.qubits,
+            clbits: self.clbits,
+            params: self.params.clone(),
+            extra_attrs: self.extra_attrs.clone(),
+            #[cfg(feature = "cache_pygates")]
+            py_op: OnceLock::new(),
```

Comment:
Toward the beginning of the discussion [here](https://github.com/Qiskit/qiskit/pull/13532#issuecomment-2521505541), @jakelishman says:

> the right solution is to manually impl Clone for PackedInstruction to not propagate the cache for gates that aren't Python-space singletons.

Should we be preserving the `py_op` in the case of singleton gates to avoid a more significant performance regression?

Reply:
I was talking mostly about what's _necessary_, rather than what's easy.  Technically it's safe to propagate the cache if the Python object itself is immutable.  If it's too hard to tell, or we deem it too risky, we can just zero them all out.

Fwiw, we'd basically just apply a Rust-only heuristic: check the `PackedOperation` is a `Standard` and it's a 0-parameter gate with `extra_attrs == None`.  That won't catch every single singleton, but it'll do the vast majority and fails safe.